lib/matplotlib/_text_layout.py

```diff
@@ -2,9 +2,15 @@
 Text layouting utilities.
 """
 
+import dataclasses
+
 from .ft2font import KERNING_DEFAULT, LOAD_NO_HINTING
 
 
+LayoutItem = dataclasses.make_dataclass(
```

Comment:
What are dataclasses getting us here that a named tuple would not?

Reply:
The thing is not iterable, which avoids accidentally relying on an unpacking API (`x, y = ...`) which would be a pain to modify (i.e. you can't add another field while maintaining a sensible deprecation).
Of course this is private API so in a sense we could do whatever we want, but as a general rule I think dataclasses are nicer (but note that I went for the "functional" declaration API because I certainly wouldn't want to be the person introducing type hints in the codebase :))

Reply:
That is a compelling agreement :+1: .